controllers/argocd/hooks.go

```diff
@@ -13,7 +13,7 @@ var (
 
 // Hook changes resources as they are created or updated by the
 // reconciler.
-type Hook func(*argoprojv1alpha1.ArgoCD, interface{}, string) error
```

Comment:
The string parameter shouldn't be removed; the `Hook` is an api that external applications can use and the string is a hint that may be used in the future to disambiguate the context of a hook call.

Reply:
You can use something like `// nolint:<category>` to suppress lint issues for specific targets. I believe this may have been a check in the `unparam` catagory, so adding a `// nolint:unparam` comment just above the type definition should silent the linter.